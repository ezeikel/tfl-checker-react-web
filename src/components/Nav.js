import React from "react";
import { NavLink } from "react-router-dom";
import styled from "styled-components";

const Wrapper = styled.nav`
  display: flex;
  align-items: center;
`;

const NavList = styled.ul`
  display: flex;
  height: 100%;
  li {
    display: flex;
    font-size: 22px;
    font-weight: bold;
    a {
      position: relative;
      color: var(--color-white);
      overflow-x: hidden;
      &:after {
        content: "";
        display: block;
        height: 3px;
        background-color: var(--color-white);
        position: absolute;
        bottom: 0;
        left: 0;
        width: 40%;
        transition: transform 0.3s ease-in-out;
        transform: translateX(-150%);
      }
      &.active {
        &:after {
          transform: translateX(0);
        }
      }
    }
  }
  li + li {
    margin-left: var(--spacing-large);
  }
`;

const Nav = () => {
  return (
    <Wrapper>
      <NavList>
        <li>
          <NavLink to="/" exact>
            Status updates
          </NavLink>
        </li>
        <li>
          <NavLink to="/trip-planner" exact>
            Plan a journey
          </NavLink>
        </li>
      </NavList>
    </Wrapper>
  );
};

export default Nav;
