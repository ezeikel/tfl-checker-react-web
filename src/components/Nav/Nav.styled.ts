import styled from "styled-components";

export const Wrapper = styled.nav`
  display: flex;
  align-items: center;
`;

export const NavList = styled.ul`
  display: flex;
  height: 100%;
  li {
    display: flex;
    font-size: 22px;
    font-weight: bold;
    a {
      position: relative;
      color: var(--color-white);
      overflow-x: hidden;
      &:after {
        content: "";
        display: block;
        height: 3px;
        background-color: var(--color-white);
        position: absolute;
        bottom: 0;
        left: 0;
        width: 40%;
        transition: transform 0.3s ease-in-out;
        transform: translateX(-150%);
      }
      &.active {
        &:after {
          transform: translateX(0);
        }
      }
    }
  }
  li + li {
    margin-left: var(--spacing-large);
  }
`;
