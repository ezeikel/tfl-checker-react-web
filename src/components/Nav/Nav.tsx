import { NavLink } from "react-router-dom";
import { Wrapper, NavList } from "./Nav.styled";

const Nav = () => {
  return (
    <Wrapper>
      <NavList>
        <li>
          <NavLink to="/">Status updates</NavLink>
        </li>
        <li>
          <NavLink to="/journey-planner">Plan a journey</NavLink>
        </li>
      </NavList>
    </Wrapper>
  );
};

export default Nav;
