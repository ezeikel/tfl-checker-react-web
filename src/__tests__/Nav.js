import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Nav from '../components/Nav';

test('renders without crashing', () => {
  const { container } = render(
    <Router>
      <Nav />
    </Router>
  );
  expect(container).toMatchSnapshot();
});
