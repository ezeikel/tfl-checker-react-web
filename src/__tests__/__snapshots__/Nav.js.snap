// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <nav
    class="sc-AxjAm iMxqKn"
  >
    <ul
      class="sc-AxirZ cBkftF"
    >
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Status updates
        </a>
      </li>
      <li>
        <a
          href="/trip-planner"
        >
          Plan a journey
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
